plugin/builtin/patchtransformer/PatchTransformer.go

```diff
@@ -51,10 +51,10 @@ func (p *plugin) Config(
 		p.Patch = string(loaded)
 	}
 
-	patchSM, errSM := h.ResmapFactory().RF().FromBytes([]byte(p.Patch))
+	patchesSM, errSM := h.ResmapFactory().RF().SliceFromBytes([]byte(p.Patch))
 	patchJson, errJson := jsonPatchFromBytes([]byte(p.Patch))
```

Comment:
I fix at https://github.com/kubernetes-sigs/kustomize/pull/5194/commits/276823744cb07c3a06c8443292aef0e5e64176a2#diff-867890f08da8fc61c6a7c1e194065fe2d018ba56bf0428e4d42cf363f52ac727R129-R132.

I found `jsonpatch.Patch` is another name of [list of `jsonpatch.Operation`](https://github.com/evanphx/json-patch/blob/50fdc0b4c9ab36933e6a0c9288fef3ac5df2b907/patch.go#L48).
I use `Go` type casting and check len.

Please check [a new test case](https://github.com/kubernetes-sigs/kustomize/pull/5194/commits/276823744cb07c3a06c8443292aef0e5e64176a2#diff-7832e3dd9c6bfd1f7c341781f44cad1387b893bda2cab51f704f2260c4b3e78fR263-R291) if you need details.

Reply:
Hi @annasong20 

Looks like multiple Json patch is working now according to [this test case](https://github.com/kubernetes-sigs/kustomize/blob/da4e8810079e2a98a3f4caad83618c53014d703b/plugin/builtin/patchtransformer/PatchTransformer_test.go#L666-L719).
Are you talking about other functions?

Reply:
Thanks for the changes @koba1t! 

I'm guessing that by ["Multiple JSON patch documents"](https://github.com/kubernetes-sigs/kustomize/issues/5049#issuecomment-1440604403), @natasha41575 was referring to multiple YAML documents that specify JSON patches? Like so,

```yaml
# file referenced under patches[].path
- op: add
  path: /some/new/path
  value: value
---
- op: replace
  path: /some/existing/path
  value: new value
```

This was my first take on her comment. I could be totally wrong. I need to take a closer look later this week.

Reply:
Hi, @koba1t! Sorry for the delay. I think I stick with my previous comment that by `Multiple JSON patch documents should not be allowed`, @natasha41575 meant that a json6902 patch in [yaml form](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/patchesjson6902/) cannot contain more than 1 yaml document. I say this for a couple of reasons

1. I could not find reference to a "json document" on the internet, but a `yaml document` is what's separated by `---` as depicted in my previous comment. Our current json6902s patching mechanism in json form accepts an array of `json object`s.
2. It would make sense that before we allow multiple yaml documents for a strategic merge patch, we want to guard multiple yaml documents for a json6902 patch.
3. From context clues, @natasha41575 says "(currently, only the first one is used and the rest are ignored. This should be an error instead.)" I tested this out locally, and if you specify multiple yaml documents like in my previous comment, indeed only the first is applied and the rest ignored. All changes to the formatting of the json form results in errors for me.

We should ask @natasha41575 next week when she's back from vacation to be sure.

Reply:
I add `TODO` comment and one test case. https://github.com/kubernetes-sigs/kustomize/pull/5194/commits/c203ca26a2420b975f0c9786687fadf009c89958
I think this problem is caused by external yaml parse lib.

Reply:
Definitely! Sounds great.